### PR TITLE
The sinatra middleware will now set the resource on rack middleware

### PR DESCRIPTION
**Pull request by jpaulgs**
This is in relation to #486

### CODE REVIEW

File: lib/ddtrace/contrib/sinatra/tracer.rb [hunk 1]
```diff
@@ -96,10 +97,14 @@ def render(engine, data, *)
...
             end
 
+            rack_span = env[Datadog::Contrib::Rack::TraceMiddleware::RACK_REQUEST_SPAN]
+
             span.resource = "#{request.request_method} #{@datadog_route}"
             span.set_tag('sinatra.route.path', @datadog_route)
             span.set_tag(Datadog::Ext::HTTP::STATUS_CODE, response.status)
             span.set_error(env['sinatra.error']) if response.server_error?
+
+            rack_span.resource ||= span.resource unless rack_span.nil?
```
Comment (delner) [inline lib/ddtrace/contrib/sinatra/tracer.rb]: By using the `||=` we aren't overriding the Rack span's resource with the presumably better Sinatra resource name, if the Rack resource is already set. Is that what we want?

Comment (irabinovitch) [reply lib/ddtrace/contrib/sinatra/tracer.rb]: @jpaulgs any thoughts?

Comment (jpaulgs) [reply lib/ddtrace/contrib/sinatra/tracer.rb]: In my application I have a few sinatra applications that run as middleware with the bottom level being the full web application.  The DD tracer is using `app.after` to set the `rack_span.resource` which I want to be set once by the lowest level that the span reached.  

I don't recall seeing the rack_span.resource being set on the way in.  If that is the case then this code would indeed need to change.

**Comment by marcotc** [comment]
👋 @jpaulgs we've just merged a PR that includes the changes in this PR.

Thank you for these changes and the detailed bug report, we'll give you credit in our changelog.

Fell free to follow the original issue (#486) or this other related issue (#913) for more details about the release of this feature.

**Pull request closed without merging**